pkg/apis/core/validation/secretbinding.go

```diff
@@ -69,6 +70,16 @@ func validateSecretReferenceOptionalNamespace(ref corev1.SecretReference, fldPat
 
 	if len(ref.Name) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath.Child("name"), "must provide a name"))
+	} else {
+		for _, err := range validation.IsDNS1123Subdomain(ref.Name) {
```

Comment:
Here, the [`ValidateSecretName`](https://github.com/kubernetes/kubernetes/blob/v1.34.0/pkg/apis/core/validation/validation.go#L291-L294) is not exposed in a staging repo. But it translates to `validation.IsDNS1123Subdomain` at the end. Hence, we are good here.